src/lib/configuration/server-configuration.ts

```diff
@@ -0,0 +1,133 @@
+const VALINTALASKENTAKERRALLA_VANHA =
+  process.env.NEXT_PUBLIC_VALINTALASKENTAKERRALLA_VANHA === 'true';
+
+export type Configuration = {
+  domain: string;
+  yleiset: Record<string, string>;
+  koodisto: Record<string, string>;
+  koutaInternal: Record<string, string>;
+  valintaperusteetService: Record<string, string>;
+  ataru: Record<string, string>;
+  valintalaskentaLaskentaService: Record<string, string>;
+  valintalaskentakoostepalvelu: Record<string, string>;
+  valintaTulosService: Record<string, string>;
+  sijoittelu: Record<string, string>;
+  valintalaskentahistoriaLinkUrl: string;
+  hakukohderyhmapalvelu: Record<string, string>;
+};
+
+export function buildConfiguration(
```

Comment:
Eikö tän nyt kannattais olla server-funktio? Eli lisää "use server;" ja "async". Silloin process.enviä pitäs pystyä lukemaan tässä funktiossa suoraan.

```suggestion
export async function buildConfiguration(
```